src/lib/menu/menu-directive.ts

```diff
@@ -65,6 +70,72 @@ export class MdMenu {
     this._showClickCatcher = bool;
   }
 
+  /**
+   * Focus the first item in the menu. This method is used by the menu trigger
+   * to focus the first item when the menu is opened by the ENTER key.
+   * TODO: internal
+   */
+  _focusFirstItem() { this.items.first.focus(); }
+
+  // TODO(kara): update this when (keydown.downArrow) testability is fixed
+  // TODO: internal
+  _handleKeydown(event: KeyboardEvent): void {
+    if (event.keyCode === DOWN_ARROW) {
+      this._focusNextItem();
+    } else if (event.keyCode === UP_ARROW) {
+      this._focusPreviousItem();
+    } else if (event.keyCode === TAB) {
+      this._handleTabKeypress(event.shiftKey);
+    }
+  }
+
+  /**
+   *  When the tab key is pressed (changing focus natively), this function syncs up
+   *  the focusedItemIndex to match the currently focused item.  If the shift key is
+   *  also pressed, we know that the focus has changed to the previous tabindex. If not,
+   *  focus has changed to the next tabindex.
+   */
+  private _handleTabKeypress(shiftPressed: boolean): void {
+    shiftPressed ? this._updateFocusedItemIndex(-1) : this._updateFocusedItemIndex(1);
+  }
+
+  /**
+   * This emits a close event to which the trigger is subscribed. When emitted, the
+   * trigger will close the menu.
+   */
+  private _emitCloseEvent(): void {
+    this._focusedItemIndex = 0;
+    this.close.emit(null);
+  }
+
+  // When focus would shift past the start or end of the menu, close the menu.
+  private _closeIfFocusLeavesMenu(): void {
+    if (this._focusedItemIndex >= this.items.length || this._focusedItemIndex < 0) {
+      this._emitCloseEvent();
+    }
+  }
+
+  private _focusNextItem(): void {
+    this._updateFocusedItemIndex(1);
+    this.items.toArray()[this._focusedItemIndex].focus();
+  }
+
+  private _focusPreviousItem(): void {
+    this._updateFocusedItemIndex(-1);
+    this.items.toArray()[this._focusedItemIndex].focus();
+  }
+
+  private _updateFocusedItemIndex(delta: number, itemArray: MdMenuItem[] = this.items.toArray()) {
```

Comment:
Need function comment, especially for the params. 

Why not just `items` or `menuItems` instead of `itemArray`?